File: backend/src/main/java/tn/company/taskManagement/entities/Project.java
package tn.company.taskManagement.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import java.util.Date;
import java.util.List;
import lombok.*;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Project {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int idProject;

    private String name;
    private String description;

    @Temporal(TemporalType.DATE)
    private Date startDate;

    @Temporal(TemporalType.DATE)
    private Date endDate;

    @OneToMany(mappedBy = "project", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private List<Task> tasks;

    @ManyToOne
    @JoinColumn(name = "manager_id")
    private ProjectManager projectManager;
}

File: frontend/src/app/models/Project.ts
import { Task } from './Task';
import { ProjectManager } from './ProjectManager';

export class Project {
  idProject: number =0;
  name: string = '';
  description: string ='';
  startDate: Date = new Date();
  endDate: Date = new Date();
  tasks: Task[] = [];
  projectManager: ProjectManager = new ProjectManager();
}
